src/api/db/models/Batch.ts

```diff
@@ -0,0 +1,348 @@
+import GraphQLError, { InternalServerError, NotFoundError } from '../../errors.js';
+import MongoPaging, { AggregationOutput } from 'mongo-cursor-pagination';
+import { WRITE_IMAGES_ROLES } from '../../auth/roles.js';
+import { randomUUID } from 'node:crypto';
+import S3 from '@aws-sdk/client-s3';
+import SQS from '@aws-sdk/client-sqs';
+import Lambda from '@aws-sdk/client-lambda';
+import { getSignedUrl } from '@aws-sdk/s3-request-presigner';
+import { DateTime } from 'luxon';
+import type * as gql from '../../../@types/graphql.js';
+import Batch, { BatchSchema } from '../schemas/Batch.js';
+import mongoose from 'mongoose';
+import BatchError from '../schemas/BatchError.js';
+import retry from 'async-retry';
+import { BaseAuthedModel, GenericResponse, MethodParams, roleCheck } from './utils-model.js';
+import { ImageErrorModel } from './ImageError.js';
+import { Context } from '../../handler.js';
+
+export class BatchModel {
+  static async queryByFilter(
+    input: gql.QueryBatchesInput,
+    context: Context,
+  ): Promise<AggregationOutput<BatchSchema>> {
+    try {
+      const pipeline: Record<'$match', Record<string, any>>[] = [
+        { $match: { user: context.user.sub } },
+        { $match: { projectId: context.user['curr_project'] } },
+      ];
+
+      if (input.filter) {
+        pipeline.push({
+          $match: {
+            processingEnd: { $exists: input.filter === 'COMPLETED' },
+          },
+        });
+      }
+
+      const result = await MongoPaging.aggregate(Batch.collection, {
+        aggregation: pipeline,
+        limit: input.limit,
+        paginatedField: input.paginatedField,
+        sortAscending: input.sortAscending,
+        next: input.next,
+        previous: input.previous,
+      });
+
+      result.results = await Promise.all(
+        result.results.map((batch) => BatchModel.augmentBatch(batch)),
+      );
+
+      return result;
+    } catch (err) {
+      if (err instanceof GraphQLError) throw err;
+      throw new InternalServerError(err as string);
+    }
+  }
+
+  static async queryById(_id: string): Promise<mongoose.HydratedDocument<BatchSchema>> {
+    const query = { _id };
+    try {
+      const batch = await Batch.findOne(query);
+      if (!batch) throw new NotFoundError('Batch not found');
+
+      BatchModel.augmentBatch(batch as Omit<BatchSchema, 'errors'>); // Avoid conflict with errors type in BatchWithErrors
```

Comment:
Odd that we don't assign the returned augmented batch from `BatchModel.augmentBatch()` to the batch variable here. I think that may have been an oversight when this was first implemented. I'll make a note to test that and correct it if it needs correcting in a different PR.

Reply:
Well the function does seem to mutate the passed in object (ie slap on a few new properties):

https://github.com/tnc-ca-geo/animl-api/blob/3f66e2faabd4376afd31e3e2da684c319aff535d/src/api/db/models/Batch.ts#L73-L75

So despite our not explicitly assigning the response back to `batch`, I think the effect is the same. 

As an aside, I do agree that it would read more clearly if we did this in a more functional-programming style where we don't mutate objects but instead create new ones.  However, I'm not entirely sure how to do that with the Mongoose models (I don't think they're plain JS objects, so generating a new one from an old one may have some surprises)...